### ISSUE TITLE
Anchor text of user pill should contain only display name and room mention the room alias or ID

### ISSUE THREAD

**Issue by bkil** [ajbura/cinny#2064]
### Describe the bug

Generated `formatted_body` in JSON is not compliant with the specification:
* https://spec.matrix.org/v1.12/client-server-api/#client-behaviour-27

### Reproduction

Send in a room mention and a user mention with tab-completion. The resulting JSON body will include the following field:

`"formatted_body": "Hello <a href=\"https://matrix.to/#/@phishbot:matrix.org\">@Phish Bot</a> come join <a href=\"https://matrix.to/#/#gemiweb0:matrix.org\">#gemiweb0, the HTML subset</a>",`

### Expected behavior

The resulting JSON body should include the following field:

`"formatted_body": "Hello <a href=\"https://matrix.to/#/@phishbot:matrix.org\">Phish Bot</a> come join <a href=\"https://matrix.to/#/#gemiweb0:matrix.org\">#gemiweb0:matrix.org</a>",`

### Platform and versions

```shell
Cinny Web v4.2.3
```

**Comment by kfiven** [comment]
The spec example shows the what you mentioned in Reproduction:
<img width="701" alt="image" src="https://github.com/user-attachments/assets/7b2fe2a7-bdef-46db-986f-1f2c2f62ce94">

Expected behaviour would not be spec compliant.

**Comment by bkil** [comment]
You are right. I have now updated the description. Note that neither the example JSON, nor the text implies that an `@` sign should be included, so Cinny is still in error in both cases.
